src/sentry/monitors/endpoints/monitor_details.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import annotations
+
+import abc
+from typing import Any
+
+from django.db import router, transaction
+from django.db.models import F
+from django.db.models.functions import TruncMinute
+from django.utils.crypto import get_random_string
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, quotas
+from sentry.api.exceptions import ParameterValidationError
+from sentry.api.helpers.environments import get_environments
+from sentry.api.serializers import serialize
+from sentry.constants import ObjectStatus
+from sentry.models.environment import Environment
+from sentry.models.project import Project
+from sentry.models.rule import Rule, RuleActivity, RuleActivityType
+from sentry.models.scheduledeletion import RegionScheduledDeletion
+from sentry.monitors.models import (
+    CheckInStatus,
+    Monitor,
+    MonitorCheckIn,
+    MonitorEnvironment,
+    MonitorStatus,
+)
+from sentry.monitors.serializers import MonitorSerializer
+from sentry.monitors.utils import (
+    create_issue_alert_rule,
+    get_checkin_margin,
+    get_max_runtime,
+    update_issue_alert_rule,
+)
+from sentry.monitors.validators import MonitorValidator
+from sentry.utils.auth import AuthenticatedHttpRequest
+from sentry.utils.outcomes import Outcome
+
+
+class MonitorDetailsMixin(abc.ABC):
+    @abc.abstractmethod
+    def create_audit_entry(self, request: Request, transaction_id=None, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def respond(self, context: object | None = None, **kwargs: Any) -> Response:
+        pass
```

Comment:
I will probably make these more generic in a followup pr, since they're inherited from the base api classes